d2/src/main/java/com/linkedin/d2/balancer/dualread/DualReadStateManager.java

```diff
@@ -188,10 +188,15 @@ public void checkAndSwitchMode(String d2ServiceName)
       LOG.info("Dual read mode executor is shut down already. Skipping getting the latest dual read mode.");
       return;
     }
-
+    if (d2ServiceName == null)
```

Comment:
when service name is null, we should the original behavior which check the global rate limiter and call updateGlobal as needed.

Reply:
Thanks for the comments. Updated